### DISCUSSION TITLE
Importance of the version of the installed libraries

### DISCUSSION THREAD

**Discussion by LaurenzV** [sebastian-hofstaetter/teaching#69] [Advanced IR: Exercise Q&A] [answered]
Hello,

Some of our group had some trouble installing the versions of the libraries as specified in the requirements.txt file, mainly because they seem to be relatively outdated (pytorch was specified as 1.6, but the most recent one is 1.11; alennlp was specified as 1.2.2, but the most recent one is 2.9.3) so they don't run on newer architectures (e.g. the M1 processor). Is it okay if we use the newest versions of these libraries or are we required to stick with the ones specified in the file?

Kind regards
Laurenz

**Comment by sebastian-hofstaetter** [comment, marked as answer]
Hi,

I would keep the specific versions (especially allennlp has lots of breaking changes between 1.x and 2.x in the dataloading, which we use). Sorry we didn't get to update the starter code in time :/ For the M1 mac: I couldn't get pytorch and the rest of the matchmaker dependencies to run natively anyway, there was always some faraway dependency not compiled for arm and trying to compile it locally would only lead you to more problems (ca. 3 months ago) - the solution is to install an intel version of (mini)conda, and this then installs all the intel compiled python versions (translated with rosetta) without any problems. It still runs very fast for local development. 

Best,
Sebastian

**Reply by mandoway** [reply]
Hello!

I also had troubles installing all dependencies on my Mac.
Would it be possible to mention some common problems (or maybe the python version used) and solutions in the assignments readme or link to them to make the lives of future students a little easier?

I will try the proposed solution now. 
Edit: works fine!
I installed the latest miniconda bash version on my 2021 mac. https://docs.conda.io/en/latest/miniconda.html#latest-miniconda-installer-links
Created a conda env in pycharm, selected python 3.6. and installed everything (took a little time though)

Since this answer is already 1yr old, is there a newer solution now? Or will it only change once the template is updated?

BR
Markus

**Comment by bruluk5w** [comment]
Hi,
I also had the same problem but it seems with the changes in this diff file it could work fine:
[commit.diff.txt](https://github.com/sebastian-hofstaetter/teaching/files/8630975/commit.diff.txt)
Can someone confirm that this is enough? The changes in the data loading were actually not that huge unless I overlooked something major.
Also, what is "matchmaker" and is it relevant for our exercise, I don't see anything related in Assignment.md?
Best,
Lukas

**Reply by LaurenzV** [reply]
I think the matchmaker is referring to this: https://github.com/sebastian-hofstaetter/matchmaker

**Comment by bruluk5w** [comment]
Thanks!
Having seen that now, @sebastian-hofstaetter  how should we deal with this information? It seems like it implements exactly large parts of our exercise, are we allowed to take it as a reference? I'm asking straight and honestly: what is the policy on plagiarism for the exercise? I'm asking because things cannot be unseen and now i have a pretty concrete idea how the network architectures are implemented in practice. I imagine it hard to implement core parts in a different way just for the sake of it.
Best regards,
Lukas

**Reply by sebastian-hofstaetter** [reply]
Hi, the matchmaker library is public information (and we are referencing to it throughout the lectures) and you are free to use it as a reference, and you may also copy parts of it. The question is how you learn best: I would recommend trying to start out from scratch, and only referencing the existing code once you are stuck, but if you work better from existing code you can take another route as well. I just caution that the matchmaker library contains a lot of complexity (inside and outside the models) for configuration and research experiments, which is not needed for this exercise. 
Best,
Sebastian
